### PR TITLE
SPCTR-2041 Fix: Block encounter an issue with react hook withSelect.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-2041
- Replaced withSelect with useSelect.
- Removed deprecated `__experimentalGetSettings` and used `getSettings` for Date.

### Screenshots

- https://share.getcloudapp.com/mXu2k8k6

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- For Column and container -
  - Drag and drop column.
  - Check variation is visible or not.
  - Add inner blocks, add a content, styling to block.
  - Verify it is visible on tablet, mobile in editor as well as frontend.
- For CF/GF and Forms
  - Drag and drop block.
  - Verify it is showing variation for block if available.
  - GF/CF Forms is rendering inside editor. Try creating new gf/cf forms and verify it is visible or not.
  - Add a child into forms blocks.
  - Try adding color to blocks.
  - Verify it is visible on tablet, mobile in editor as well as frontend.
- For Post block 
  - Drag and drop Grid/Masonry/Carousel/Post timeline.
  - Try changing post type and verify post are showing in block as per post type set.
  - Change Taxonomy type to categories, Tags and verify posts are showing accordingly.
  - Post Per Page, Columns and content setting.
  - Try adding colors to post.
- For Taxonomy
  - Drag and drop block.
  - Make sure you have different tags categories for post.
  - Verify according to tags and category taxonomy showing or not.
  - Try different layout, add a styling to block.
- For Table of content
  - Add a different heading on page and verify they are visible inside editor as well as on front or not.
  - Try different styling to TOC block.
- How-To
  - Drag and drop How-To
  - Edit the content.
  - Check schema is generated in JSON file by inspecting block.
  - Verify schema as per content added.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/how-to/edit.js [hunk 1]
```diff
@@ -118,105 +223,4 @@ const HowToComponent = ( props ) => {
...
-			}
-			jsonData.step[key] = stepsData;
-		} );
-
-		return {
-			schemaJsonData: jsonData,
-		};
-	} )
-)( HowToComponent );
+export default compose()( HowToComponent );
```
Comment (RajkiranBagal) [inline src/blocks/how-to/edit.js:226]: Is compose needed here? If yes then Why?

File: src/blocks/post/post-grid/edit.js [hunk 2]
```diff
@@ -247,129 +345,23 @@ const PostGridComponent = ( props ) => {
...
 				setStateValue={ setStateValue }
 				togglePreview={ togglePreview }
+				latestPosts={ latestPosts }
 			/>
 			</>
 
 	);
 };
 
 export default compose(
```
Comment (RajkiranBagal) [inline src/blocks/post/post-grid/edit.js:364]: Is compose needed here?

File: src/blocks/post/post-masonry/edit.js [hunk 3]
```diff
@@ -2545,110 +2637,23 @@ const UAGBPostMasonry = ( props ) => {
...
 				togglePreview={ togglePreview }
+				latestPosts={ latestPosts }
+				categoriesList={ categoriesList }
 			/>
 			</>
 
 	);
 };
 
 export default compose(
```
Comment (RajkiranBagal) [inline src/blocks/post/post-masonry/edit.js:2656]: Is compose needed here?

File: src/blocks/table-of-contents/edit.js [hunk 4]
```diff
@@ -250,112 +353,10 @@ const UAGBTableOfContentsEdit = ( props ) => {
...
-				}
-			} );
-		}
-
-		return {
-			headers,
-		};
-	} )
-)( UAGBTableOfContentsEdit );
+export default compose()( UAGBTableOfContentsEdit );
```
Comment (RajkiranBagal) [inline src/blocks/table-of-contents/edit.js:362]: Is compose needed here?

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/how-to/edit.js (hunk 1 above)
Comment (AbhijitNage123) [reply src/blocks/how-to/edit.js]: Resolved

File: src/blocks/post/post-grid/edit.js (hunk 2 above)
Comment (AbhijitNage123) [reply src/blocks/post/post-grid/edit.js]: Resolved

File: src/blocks/post/post-masonry/edit.js (hunk 3 above)
Comment (AbhijitNage123) [reply src/blocks/post/post-masonry/edit.js]: Resolved

File: src/blocks/table-of-contents/edit.js (hunk 4 above)
Comment (AbhijitNage123) [reply src/blocks/table-of-contents/edit.js]: Resolved

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**